store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreInputChainTest.java

```diff
@@ -0,0 +1,131 @@
+    public static final String INPUT_STRING = "AbCd";
+    private FederatedStore federatedStore;
+
+    @AfterAll
+    public static void after() {
+        resetForFederatedTests();
+    }
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        resetForFederatedTests();
+
+        federatedStore = new FederatedStore();
+        federatedStore.initialise(GRAPH_ID_TEST_FEDERATED_STORE, null, new FederatedStoreProperties());
+
+        federatedStore.execute(
+                new AddGraph.Builder()
+                        .graphId(GRAPH_ID_A)
+                        .storeProperties(loadAccumuloStoreProperties(ACCUMULO_STORE_SINGLE_USE_PROPERTIES))
+                        .schema(basicEntitySchema())
+                        .isPublic(true)
+                        .build(), contextBlankUser());
+    }
+
+    @Test
+    public void shouldCorrectlySetGenericInputInFederatedOperationChain() throws OperationException {
+        // Given / When
+        final Object output = federatedStore.execute(
+                new OperationChain.Builder()
+                        .first(new SetVariable.Builder()
+                                .input(INPUT_STRING)
+                                .variableName(VAR)
+                                .build())
+                        .then(new GetVariable.Builder()
+                                .variableName(VAR)
+                                .build())
+                        .then(new FederatedOperation.Builder()
+                                .op(new If.Builder()
+                                        .conditional(new Conditional(new IsA(STRING)))
+                                        .then(new Map<>(Lists.newArrayList(new ToUpperCase(), new ToList())))
+                                        .otherwise(new Map<>(Lists.newArrayList(new ToLowerCase(), new ToList())))
+                                        .build())
+                                .build())
+                        .build(), contextBlankUser());
+
+        // Then
+        assertThat(output)
+                .isInstanceOf(List.class)
+                .isEqualTo(Lists.newArrayList(INPUT_STRING.toUpperCase()));
+    }
+
+    @Test
+    public void shouldCorrectlySetMultiInputInFederatedOperationChain() throws OperationException {
+        // Given / When
+        final Iterable<? extends Element> output = federatedStore.execute(
+                new OperationChain.Builder()
+                        .first(new SetVariable.Builder()
+                                .input(Lists.newArrayList(entityBasic()))
+                                .variableName(VAR)
+                                .build())
+                        .then(new GetVariable.Builder()
+                                .variableName(VAR)
+                                .build())
+                        .then(new FederatedOperation.Builder()
+                                .op(new AddElements())
+                                .build())
```

Comment:
Yes, this would be a normal operation chain then, this test is specifically for inputs chaining into FederatedOperation. I can make another test though?

Reply:
In that case I'd suggest the test have a different name, something like `shouldCorrectlySetMultiInputInOperationChainWithFederatedOperation`. There's no need for another test really, that it goes to all graphs is tested elsewhere.